plugins/experimental/wasm/CMakeLists.txt

```diff
@@ -18,5 +18,28 @@
 add_subdirectory(lib)
 
 add_atsplugin(wasm ats_context.cc ats_wasm.cc wasm_main.cc)
-target_link_libraries(wasm PRIVATE wasmlib wamr::wamr)
+
+set(WASM_RUNTIME wasmlib)
+if(wamr_FOUND)
+  list(APPEND WASM_RUNTIME wamr::wamr)
+endif()
+if(wasmtime_FOUND)
+  list(APPEND WASM_RUNTIME wasmtime::wasmtime)
+endif()
+if(wasmedge_FOUND)
+  list(APPEND WASM_RUNTIME wasmedge::wasmedge)
+endif()
```

Comment:
If having multiple runtime included might be an issue, you could use else if to ensure only one. You'll have to pick some priorities to the libraries, but users could still select their preference with cmake directives.

Reply:
yes. you are right. wasmtime and wamr cannot be used together.